### PR TITLE
CommandPalette: Add actions for entire Nav tree

### PR DESCRIPTION
**Pull request by joshhunt**
**What is this feature?**

Exposes the entire nav tree in the command palette so all plugins and pages can be navigated to via command palette.

image

**Why do we need this feature?**

Adding this as a (power-user) way to quickly get to pages deeper in the nav stack in feedback from topnav changes

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer**:

still work in progress, i would at least like to clean up the navtree -> actions code, and maybe make it properly recursive.

also im not sure if the _entire_ nav tree (pages that we don't show directly in the side nav) are visible yet?

### CODE REVIEW

File: public/app/features/commandPalette/actions/global.static.actions.ts
Comment (joshhunt) [inline public/app/features/commandPalette/actions/global.static.actions.ts:97]: Putting `perform` on parent nav tree actions (like "Go to Administration") in the screenshot prevent the user from being able to view that item's child actions in the command palette because instead it will navigate to them. I think that's fine?

**Review by ashharrison90 — COMMENTED** [review]
this looks awesome! 🙌 

couple of notes:
- now that administration is split into sections, think we should just make the flatmap fully recursive and not limit it to 1 level of depth
- seeing `Go to` multiple times is confusing, maybe we could only show it for the item without a parent?

this gives something like 
image

side note: the performance problems i was seeing previously with command palette are completely to do with searching dashboards in it. i have a dashboard with a long name (`lorem ipsum etc...`) that causes it to lock up completely.

**Comment by joshhunt** [comment]
all great feedback @ashharrison90

also check out Github's cmd-k bar for more ideas for how to structure. I like having a section that is just "Pages".

I also like how they have nav items that can be navigated to by hitting [Enter], or you hit [Tab] to traverse into and switch between those pages. Food for thought.


4081_2022-11-28-17-14_chrome
nt.com/46142/204340235-819d19ea-e151-44ec-aae4-55f4406dd970.png)

File: public/app/features/commandPalette/ResultItem.tsx
Comment (ashharrison90) [inline public/app/features/commandPalette/ResultItem.tsx:51]: hyper minor nit: `index` is unused

File: public/app/features/commandPalette/ResultItem.tsx
Comment (ashharrison90) [inline public/app/features/commandPalette/ResultItem.tsx:106]: nice spot 👀

**Review by ashharrison90 — APPROVED** [review]
think this looks great - worth waiting for a review from the peeps who originally implemented command palette tho 🙌

File: public/app/features/commandPalette/actions/global.static.actions.tsx
Comment (gelicia) [inline public/app/features/commandPalette/actions/global.static.actions.tsx:47]: :D

**Review by gelicia — APPROVED** [review]

**Pull request merged**
